receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_scraper_linux.go

```diff
@@ -34,6 +38,18 @@ func (s *memoryScraper) recordLinuxMemoryAvailableMetric(now pcommon.Timestamp,
 	s.mb.RecordSystemLinuxMemoryAvailableDataPoint(now, int64(memInfo.Available))
 }
 
+func (s *memoryScraper) recordLinuxMemoryDirtyMetrics(now pcommon.Timestamp, memInfo *mem.VirtualMemoryStat) {
+	// This value is collected from /proc/meminfo and converted from kB to bytes in gopsutil:
+	// https://github.com/shirou/gopsutil/blob/d8750909ba41f2de9750c90a6d2074c68dfc677e/mem/mem_linux.go#L148
+	dirtyBytes := int64(memInfo.Dirty)
+	s.mb.RecordSystemLinuxMemoryDirtyDataPoint(now, dirtyBytes)
+	// The original kB number from /proc/meminfo is a calculation of the number of dirty pages
+	// converted to kB based on the PAGE_SIZE.
+	// https://gist.github.com/braydonk/78ff4581be2056c0b3e77b471088258a
+	s.mb.RecordSystemLinuxMemoryDirtyPageCountDataPoint(now, dirtyBytes/pageSize)
```

Comment:
We don't have any other metrics reporting number of pages, right? Why would we do that only for this one specifically? Should the page size be reported separately instead?

Reply:
This value is likely to be used for calculation on the backend (or transform processor). It'll be hard to achieve with a resource attribute. We have a few metrics that emit constant value because of the same reasons. See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sclusterreceiver/metadata.yaml#L221 for example

Reply:
Okay, if there's precedent for it then I'm fine with it. I'll make the change now.

Reply:
Done in [2611c50](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38673/commits/2611c506c4bc6a84c2f78a94acfd9f82952d082a)